### ISSUE TITLE
JAVA_HOME set wrong in Dockerfiles

### ISSUE THREAD

**Issue by mmccarthy404** [corretto/corretto-docker#129]
In the corretto Dockerfiles, `JAVA_HOME` appears to be set wrong. For example here:

https://github.com/corretto/corretto-docker/blob/main/17/headful/al2022/Dockerfile#L24

Here we see that `JAVA_HOME=/usr/lib/jvm/java-17-amazon-corretto`. However, when building an image from this image, I get the error:
```
ERROR: JAVA_HOME is set to an invalid directory: /usr/lib/jvm/java-17-amazon-corretto

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

I believe that JAVA_HOME should be set to `/usr/lib/jvm/java-17-amazon-corretto.$(uname -m)`. Which fixes this issue.

**Comment by earthling-amzn** [comment]
Thank you for reporting this, we will look into it.

**Comment by lutkerd** [comment]
I don't believe we can call a program or bash from the ENV line and the proper fix is likely to move creation of the alternative path from the [devel package](https://github.com/corretto/corretto-17/blob/develop/installers/linux/al2/spec/java-amazon-corretto.spec.template#L284) to the headless package. This will be done prior to GA of the new version of Amazon Linux.

**Comment by kymppi** [comment]
Is this issue still suppose to be open?

**Comment by lutkerd** [comment]
This is resolved.
